src/cost_model.rs

```diff
@@ -57,6 +57,29 @@ pub fn estimate_cycles(i: Instruction) -> u64 {
         insts::OP_REMW => 32,
         insts::OP_REMU => 32,
         insts::OP_REMUW => 32,
+        // A
+        insts::OP_LR_W => 4,
+        insts::OP_SC_W => 4,
+        insts::OP_AMOSWAP_W => 7,
+        insts::OP_AMOADD_W => 7,
+        insts::OP_AMOXOR_W => 7,
+        insts::OP_AMOAND_W => 7,
+        insts::OP_AMOOR_W => 7,
+        insts::OP_AMOMIN_W => 7,
+        insts::OP_AMOMAX_W => 7,
+        insts::OP_AMOMINU_W => 7,
+        insts::OP_AMOMAXU_W => 7,
+        insts::OP_LR_D => 3,
+        insts::OP_SC_D => 3,
+        insts::OP_AMOSWAP_D => 5,
```

Comment:
I was slightly confused, how is this 5 calculated?

Reply:
OP_AMOSWAP_D contains a memory load(2 cycles) and a memory store(2 cycles), and the swap operation takes 1 cycles

Reply:
Okay I see this now but honestly I have slightly different opinions. In CPU's perspective, each of the atomic operations essentially only *touch* one memory location, yes it is one read and one write but the time that matters, is the time to load the memory location into CPU cache, which happens only once here. Rest work is simply touching cache and can basically be counted as free. So in my opinions, this should be 1 cycle for calculation + 2 cycles for memory operations, which is 3 cycles.